db/functions.go

```diff
@@ -90,10 +98,52 @@ func (db *Database) CallUserFunction(name string, args map[string]interface{}, m
 
 // Top-level public method to run a GraphQL query on a db.Database.
 func (db *Database) UserGraphQLQuery(query string, operationName string, variables map[string]interface{}, mutationAllowed bool, ctx context.Context) (*graphql.Result, error) {
-	if graphql := db.Options.GraphQL; graphql != nil {
-		return db.Options.GraphQL.Query(db, query, operationName, variables, mutationAllowed, ctx)
-	} else {
+	if graphql := db.Options.GraphQL; graphql == nil {
 		return nil, base.HTTPErrorf(http.StatusServiceUnavailable, "GraphQL is not configured")
+	} else {
+		return graphql.Query(db, query, operationName, variables, mutationAllowed, ctx)
+	}
+}
+
+//////// UTILITIES
+
+// Returns an HTTP 413 error if `maxSize` is non-nil and less than `actualSize`.
+func CheckRequestSize[T int | int64](actualSize T, maxSize *int) error {
+	if maxSize != nil && int64(actualSize) > int64(*maxSize) {
+		return base.HTTPErrorf(http.StatusRequestEntityTooLarge, "Arguments too large")
+	} else {
+		return nil
+	}
+}
+
+// Utility that returns a rough estimate of the original size of the JSON a value was parsed from.
+func EstimateSizeOfJSON(args any) int {
+	switch arg := args.(type) {
+	case bool:
+		return 4
+	case int64:
+		return 4
```

Comment:
8 bytes?

Reply:
Nope; this function roughly estimates the length of the JSON encoding, not the size of a binary value. The string `true` is 4 bytes and `false` is 5; I simplified to just return 4. An integer is variable length, but rather than having to do logarithms or build a string, I'm just handwaving 4 as being a conservative average length.

This function is admittedly sloppy (it's just for detecting cases where the size goes crazy, likely due to abuse), but it needs to be fast because it's going to be run before every function call.